collections/sort_by.ts

```diff
@@ -24,26 +24,16 @@
  */
 export function sortBy<T>(
   array: Array<T>,
-  selector:
-    | ((el: T) => number)
-    | ((el: T) => string)
-    | ((el: T) => bigint)
-    | ((el: T) => Date),
+  selector: ((el: T) => number | string | bigint | Date),
```

Comment:
By this change, the selector's return value can be mixed types. I don't think that is better design than the current one. (For example, we can provide a selector like `(item) => cond(item) ? "foo" : 1)`, which doesn't make much sense.

Reply:
Ah, fair enough. I can revert that. Thanks for the tip!

Reply:
Why don't you use type Selector ?

Reply:
@Ayfri good suggestion. Made that change as well.